OpenRA.Mods.Common/Traits/Render/WithDeathAnimation.cs

```diff
@@ -70,10 +74,33 @@ public WithDeathAnimation(Actor self, WithDeathAnimationInfo info)
 
 		public void Killed(Actor self, AttackInfo e)
 		{
-			// Killed by some non-standard means. This includes being crushed
-			// by a vehicle (Actors with Crushable trait will spawn CrushedSequence instead).
+			// Actors with Crushable trait will spawn CrushedSequence.
+			if (crushed)
+			{
+				if (Info.CrushedSequence != null)
```

Comment:
You could move this into the OnCrush handler, and then just have `if (crushed) return;`